adagrams/game.py

```diff
@@ -1,13 +1,129 @@
+    
+    for letter in word:
+        if letter.upper() in list_of_available_letters:
+            
+            # list_of_available_letters.remove(letter.upper())
+            
+            # alternative approach without using remove():
+            for i in range(len(list_of_available_letters)):
+                if list_of_available_letters[i] == letter.upper():
+                    list_of_available_letters[i] = 0
+                    break
+        else:
+            return False
+        
+    return True
 
 def score_word(word):
-    pass
+    """
+    Returns the score of a given word as defined by the Adagrams game.
+    
+    Parameters:
+        word (string): string of characters.
+
+    Returns:
+        int: number of points.
+    """
+    # Each letter's point value is described in the table below
+    score_chart = { "A": 1, "B": 3, "C": 3, "D": 2, "E": 1, "F": 4, "G": 2, "H": 4, "I": 1, "J": 8, "K": 5, "L": 1, "M": 3, "N": 1, "O": 1, "P": 3, "Q": 10, "R": 1, "S": 1, "T": 1, "U": 1, "V": 4, "W": 4, "X": 8, "Y": 4, "Z": 10 }
+    
+    score = 0
+
+    # Each letter within word has a point value. The number of points of each letter is summed up to represent the total score of word
+    for letter in word:
+        if letter.upper() in score_chart:
+            score += score_chart[letter.upper()]
+    if len(word) >= 7:
+                score += 8 # If the length of the word is 7, 8, 9, or 10, then the word gets an additional 8 points
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    This function looks at the list of word_list and calculates which of these words has the highest score.
+    
+    Parameters:
+        word_list (list): list of strings.
+
+    Returns:
+        tuple: a winning word and it's score.
+    """
+
+    highest_score = 0
+    pool_of_words_with_scores = {}
+    for word in word_list:
+        score = score_word(word)
+        pool_of_words_with_scores[word] = score
+
+    for word, score in pool_of_words_with_scores.items():
+        if score > highest_score:
+            highest_score = score
+    
+    # In the case of tie in scores:
+    # prefer the word with the fewest letters...
+    # ...unless one word has 10 letters. If the top score is tied between multiple words and one is 10 letters long, choose the one with 10 letters over the one with fewer tiles
+    # If the there are multiple words that are the same score and the same length, pick the first one in the supplied list
+    min_length = 11
+    for word, score in pool_of_words_with_scores.items():
+        if len(word) == 10 and score == highest_score:
+            winning_word = word
+            break
+        elif len(word) < min_length and score == highest_score:
+            min_length = len(word)
+            winning_word = word
+        
+    # index 0 ([0]): a string of a word
+    # index 1 ([1]): the score of that word
+    return (winning_word, highest_score)
```

Comment:
It's pretty common to see a tuple written without parens, like:
```suggestion
    return winning_word, highest_score
```